node/pkg/watchers/evm/connectors/batch_poller.go

```diff
@@ -61,48 +58,50 @@ func NewBatchPollConnector(ctx context.Context, logger *zap.Logger, baseConnecto
 		batchData:    batchData,
 		generateSafe: !safeSupported,
 	}
-	err := supervisor.Run(ctx, "batchPoller", common.WrapWithScissors(connector.runFromSupervisor, "batchPoller"))
-	if err != nil {
-		return nil, err
-	}
-	return connector, nil
+
+	return connector
 }
 
 func (b *BatchPollConnector) SubscribeForBlocks(ctx context.Context, errC chan error, sink chan<- *NewBlock) (ethereum.Subscription, error) {
-	sub := NewPollSubscription()
-	blockSub := b.blockFeed.Subscribe(sink)
-
-	// The feed library does not support error forwarding, so we're emulating that using a custom subscription and
-	// an error feed. The feed library can't handle interfaces which is why we post strings.
-	innerErrSink := make(chan string, 10)
-	innerErrSub := b.errFeed.Subscribe(innerErrSink)
-
 	// Use the standard geth head sink to get latest blocks. We do this so that we will be notified of rollbacks. The following document
 	// indicates that the subscription will receive a replay of all blocks affected by a rollback. This is important for latest because the
 	// timestamp cache needs to be updated on a rollback. We can only consider polling for latest if we can guarantee that we won't miss rollbacks.
 	// https://ethereum.org/en/developers/tutorials/using-websockets/#subscription-types
 	headSink := make(chan *ethTypes.Header, 2)
-	_, err := b.Connector.SubscribeNewHead(ctx, headSink)
+	headerSubscription, err := b.Connector.SubscribeNewHead(ctx, headSink)
 	if err != nil {
-		return nil, fmt.Errorf("failed to subscribe for latest blocks: %w", err)
+		return headerSubscription, fmt.Errorf("failed to subscribe for latest blocks: %w", err)
 	}
 
-	// Use the poller for finalized and safe.
+	// Get the initial blocks.
+	lastBlocks, err := b.getBlocks(ctx, b.logger)
+	if err != nil {
+		b.logger.Error("failed to get initial blocks", zap.Error(err))
+		return headerSubscription, fmt.Errorf("failed to get initial blocks: %w", err)
+	}
+
+	errCount := 0
+
 	common.RunWithScissors(ctx, errC, "block_poll_subscribe_for_blocks", func(ctx context.Context) error {
+		timer := time.NewTimer(b.Delay)
+		defer timer.Stop()
 		for {
 			select {
 			case <-ctx.Done():
-				blockSub.Unsubscribe()
-				innerErrSub.Unsubscribe()
-				return nil
-			case <-sub.quit:
-				blockSub.Unsubscribe()
-				innerErrSub.Unsubscribe()
-				sub.unsubDone <- struct{}{}
 				return nil
-			case v := <-innerErrSink:
-				b.logger.Info("batch_poller innerErrSink received an error, forwarding it on", zap.String("v", v))
-				sub.err <- fmt.Errorf(v)
+			case <-timer.C:
+				lastBlocks, err = b.pollBlocks(ctx, sink, lastBlocks)
+				if err != nil {
+					errCount++
+					b.logger.Error("batch polling encountered an error", zap.Int("errCount", errCount), zap.Error(err))
+					if errCount > 3 {
+						errC <- fmt.Errorf("polling encountered too many errors: %w", err)
```

Comment:
should this return the error so that runWithScissors can handle the channel publish?

Reply:
I believe the pattern in the watchers is that the go routines should post to errC and return nil. Isn't that right @panoel?